src/gwt/src/org/rstudio/core/client/AnsiCode.java

```diff
@@ -421,25 +467,45 @@ else if (codeVal == STRIKETHROUGH_OFF)
          }
          else if (Color.isFgColorCode(codeVal))
          {
-            currentColor_.setCode(codeVal);
-            resetForeground();
-            clazzes_.add(clazzForColor(codeVal));
+            if (!inverted_)
+               setForegroundColor(codeVal);
+            else
+               setBackgroundColor(Color.fgToBgColor(codeVal));
          }
          else if (Color.isBgColorCode(codeVal))
          {
-            currentBgColor_ = new Color(false, codeVal);
-            resetBackground();
-            clazzes_.add(clazzForBgColor(codeVal));
+            if (!inverted_)
+               setBackgroundColor(codeVal);
+            else
+               setForegroundColor(Color.bgToFgColor(codeVal));
          }
          else if (codeVal == RESET_FOREGROUND)
          {
-            currentColor_.reset();
-            resetForeground();
+            if (!inverted_)
+            {
+               currentColor_.reset();
+               resetForeground();
+            }
+            else
+            {
+               currentBgColor_.reset();
+               resetBackground();
+               clazzes_.add(INVERSE_BG_STYLE);
+            }
          }
          else if (codeVal == RESET_BACKGROUND)
          {
-            currentBgColor_.reset();
-            resetBackground();
+            if (!inverted_)
```

Comment:
Totally up to you but I find these statements easier to read without the negation (i.e. `if (inverted_) { ... `)

Reply:
I actually had it that way originally but I  kept misreading it (my brain was expecting the “normal” case to be first, it seemed).